### PR TITLE
Adding comments rest API implementation to the new store v1.0 rest API

### PR DESCRIPTION
**Pull request by chanaka3d**
## Purpose
Copying the document related rest API implementation content for Store v1.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: old license year

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import org.wso2.carbon.apimgt.rest.api.store.v1.dto.DocumentDTO;
+import org.wso2.carbon.apimgt.rest.api.store.v1.dto.DocumentListDTO;
+import org.wso2.carbon.apimgt.rest.api.util.RestApiConstants;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** This class is responsible for mapping APIM core documentation related objects into REST API documentation
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: need to format the comment

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+ *  related DTOs 
+ *
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: extra line

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+ */
+public class DocumentationMappingUtil {
+
+    /** Converts a APIM core Document object into corresponding REST API Document DTO object
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: please format the comment

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+     * @return a new DocumentDTO object corresponding to given Documentation object
+     */
+    public static DocumentDTO fromDocumentationToDTO(Documentation documentation) {
+        DocumentDTO documentDTO = new DocumentDTO();
+        documentDTO.setDocumentId(documentation.getId());
+        documentDTO.setName(documentation.getName());
+        documentDTO.setSummary(documentation.getSummary());
+        documentDTO.setType(DocumentDTO.TypeEnum.valueOf(documentation.getType().toString()));
+        documentDTO.setOtherTypeName(documentation.getOtherTypeName());
+        if (documentation.getSourceType() != null)
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: we need to add the curly braces here

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        int start = offset < documentations.size() && offset >= 0 ? offset : Integer.MAX_VALUE;
+        int end = offset + limit - 1 <= documentations.size() - 1 ? offset + limit - 1 : documentations.size() - 1;
+        for (int i = start; i <= end; i++) {
+            documentDTOs.add(fromDocumentationToDTO(documentations.get(i)));
+        }
+        documentListDTO.setCount(documentDTOs.size());
+        return documentListDTO;
+    }
+
+    /** Sets pagination urls for a DocumentListDTO object given pagination parameters and url parameters
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: formatting

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml [hunk 7]
```diff
@@ -355,6 +355,7 @@ paths:
       parameters:
         - $ref: '#/parameters/apiId'
         - $ref: '#/parameters/If-None-Match'
+        - $ref : '#/parameters/requestedTenant'
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: need to generate java source files.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java (hunk 2 above)
Comment (chanaka3d) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java (hunk 4 above)
Comment (chanaka3d) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java (hunk 6 above)
Comment (chanaka3d) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/DocumentationMappingUtil.java]: Fixed

**Comment by dushansilva** [comment]
https://github.com/wso2/product-apim/issues/4738

**Pull request merged**